cat-launcher/src/pages/ModsPage/ModsList.tsx

```diff
@@ -1,34 +1,98 @@
-import { useQuery } from "@tanstack/react-query";
+import {
+  useMutation,
+  useQuery,
+  useQueryClient,
+} from "@tanstack/react-query";
+import { useEffect, useState } from "react";
 
 import { SearchInput } from "@/components/SearchInput";
 import type { GameVariant } from "@/generated-types/GameVariant";
-import { listAllMods } from "@/lib/commands";
+import type { Mod } from "@/generated-types/Mod";
+import type { ModsUpdatePayload } from "@/generated-types/ModsUpdatePayload";
+import { useSearch } from "@/hooks/useSearch";
+import {
+  listenToModsUpdate,
+  triggerFetchModsForVariant,
+} from "@/lib/commands";
 import { queryKeys } from "@/lib/queryKeys";
+import { setupEventListener, toastCL } from "@/lib/utils";
 import ModCard from "./ModCard";
-import { useEffect } from "react";
-import { toastCL } from "@/lib/utils";
-import { useSearch } from "@/hooks/useSearch";
 
 interface ModsListProps {
   variant: GameVariant;
 }
 
 export default function ModsList({ variant }: ModsListProps) {
-  const {
-    data: mods,
-    isLoading,
-    error,
-  } = useQuery({
+  const queryClient = useQueryClient();
```

Comment:
Combine the hook calls related to receiving and updating mods list into a single hook.